src/Generator/CreateTargetStatementsGenerator.php

```diff
@@ -109,10 +109,6 @@ private function targetAsStdClass(GeneratorMetadata $metadata): ?Stmt
      */
     private function constructorArguments(GeneratorMetadata $metadata): array
     {
-        if (!$metadata->isTargetUserDefined()) {
-            return [];
-        }
-
```

Comment:
I don't understand why there is it condition. It was introduced by https://github.com/jolicode/automapper/commit/1486a85f666d0d83fbd8d9806ccd4f54cea97b97#diff-538038bd35959095e72795c2a102193df6a1ee0c17b27dd7f129e23d8f2464ecR111-R113
This prevent instanciating `new ArrayObject`.